ocs_ci/helpers/ocs_metrics_exporter_helpers.py

```
@@ -0,0 +1,245 @@
+    Args:
+        metrics_text (str): raw Prometheus text from /metrics
+
+    Returns:
+        dict: {metric_name: [{"labels": {...}, "value": str}, ...]}
+    """
+    families = {}
+    sample_re = re.compile(
+        r"^([a-zA-Z_:][a-zA-Z0-9_:]*)(?:\{([^}]*)\})?\s+(\S+)(?:\s+\S+)?$"
+    )
+    for line in metrics_text.splitlines():
+        line = line.strip()
```

Ruff (0.16.0)

[warning] `for` loop variable `line` overwritten by assignment target

(PLW2901)